Purchases/Purchasing/StoreKitAbstractions/PromotionalOffer.swift

```diff
@@ -14,17 +14,49 @@
 import Foundation
 import StoreKit
 
-/// The signed discount applied to a payment.
-/// Contains the details of a promotional offer discount that you want to apply to a payment.
-internal struct PromotionalOffer {
-    var identifier: String
-    var keyIdentifier: String
-    var nonce: UUID
-    var signature: String
-    var timestamp: Int
+/// Represents a ``StoreProductDiscount`` that has been validated and
+/// is ready to be used for a purchase.
+///
+/// #### Related Symbols
+/// - ``Purchases/getPromotionalOffer(forProductDiscount:product:)``
+/// - ``Purchases/getPromotionalOffer(forProductDiscount:product:completion:)``
+/// - ``StoreProduct/getEligiblePromotionalOffers()``
+/// - ``Purchases/getEligiblePromotionalOffers(forProduct:)``
+/// - ``Purchases/purchase(package:promotionalOffer:)``
+/// - ``Purchases/purchase(package:promotionalOffer:completion:)``
+/// - ``Purchases/purchase(product:promotionalOffer:)``
+/// - ``Purchases/purchase(product:promotionalOffer:completion:)``
+@objc(RCPromotionalOffer)
+public final class PromotionalOffer: NSObject {
+
+    /// The ``StoreProductDiscount`` in this offer.
+    public let discount: StoreProductDiscount
+    let signedData: SignedData
+
+    init(discount: StoreProductDiscountType, signedData: SignedData) {
+        self.discount = StoreProductDiscount.from(discount: discount)
+        self.signedData = signedData
+    }
+
 }
 
+// MARK: - SignedData
+
 extension PromotionalOffer {
+
+    /// Contains the details of a promotional offer discount that you want to apply to a payment.
+    internal struct SignedData {
```

Comment:
Note that this is still an implementation detail that doesn't need to get exposed as public 🥳